sklearn/manifold/_isomap.py

```diff
@@ -180,8 +189,16 @@ def __init__(
         self.metric_params = metric_params
 
     def _fit_transform(self, X):
+        if self.n_neighbors is not None and self.radius is not None:
+            warnings.warn(
+                "Radius is ignored when both n_neighbors and radius "
+                "are provided, n_neighbors should be set to None "
+                "explicitly when radius is passed."
```

Comment:
Good catch, I agree this is counter intuitive and we should ignore `n_neighbors`.

I am also ok raising an error instead (see https://github.com/scikit-learn/scikit-learn/pull/19794#discussion_r652488449), but then the same problem would arise, because `Isomap(radius=0.8)` would raise an error. To solve this issue, we could have both default to None, and if none are provided use `n_neighbors=5`. WDYT ?

Reply:
If we error, I was thinking of letting `Isomap(radius=0.8)` error and recommending `Isomap(radius=0.8, n_neighbors=None)`.

> To solve this issue, we could have both parameters default to None, and if none are provided use n_neighbors=5. WDYT ?

I am +0.75 with this. It is the only solution we have to make `Isomap(radius=0.8)` work. The downside is that one needs to read the docstring to get the default value. Explaining the behavior in the docstring is slightly move involved:

```python
n_neighbors : If `None` and `radius` is `None`, then the default is 5.
radius :  Can be set only when `n_neighbors` is None.
```

Reply:
Recommending `Isomap(radius=0.8, n_neighbors=None)` is a bit heavier, but I agree it is simpler to explain/understand and more explicit than a default hidden in the docstring. Let's do that.

Reply:
I've updated the code to raise a ValueError when both arguments are set, might need some help with the phrasing of the error message